fossas/fossa-cli
# [ANE-492] - Documents subproject filtering capabilities (for gradle, etc.)

# Overview

Documents subproject filtering via target.

## Testing plan

1) Get https://docs.gradle.org/current/samples/zips/sample_building_java_applications_multi_project-kotlin-dsl.zip
2) Update `utilities/build.gradle.kts` to following:

```kts
plugins {
    id("demo.java-library-conventions")
}

dependencies {
    api(project(":list"))

    // Adding dependency for testing
    // https://mvnrepository.com/artifact/org.flywaydb/flyway-core
    implementation("org.flywaydb:flyway-core:9.3.0")
}
```
3) Run `fossa analyze -o | jq`, and observe that `mvn+org.apache.commons:commons-text$1.9` is included from `:app`
4) Create `.fossa.yml`
```yml
version: 3

targets:
  only:
    - type: gradle
      path: ./
      target: ':utilities'
```
5) Run `fossa analyze -o | jq`, and observe that `mvn+org.apache.commons:commons-text$1.9` is not included, only `mvn+org.flywaydb:flyway-core$9.3.0`

## Risks

N/A

## References

https://teamfossa.slack.com/archives/C0155DTGWB1/p1660151930784179?thread_ts=1660142390.251299&cid=C0155DTGWB1
https://fossa.atlassian.net/browse/ANE-492

## Checklist
